### DISCUSSION TITLE
Reusing a watcher instance, storing the watched array and manipulate it

### DISCUSSION THREAD

**Discussion by jcharnley** [paulmillr/chokidar#1362] [Q&A]
Just a quick question, can we reuse a watcher instance? Sometimes the watcher is attached to very large network drive projects (paths) which can be upward of 100GB in size.

Our app creates a workspace in relation to the project path and we create loads of document databases which becomes a workspace. Users can switch workspaces which change the database. The first chokidar scan can take a while to become ready, which is fine, but on each change, it will do a full rescan of the chokidar watched folder. This can take a lot of time, so I was thinking, can we reuse what it was the last time it was initialized? If I went down this route, I wondered if it is possible to manipulate the watched array to add any changes that have happened when the watcher isn’t attached to that path.

I can store the watched in a document database and request it but wasn’t sure if it’s a good idea to start messing with the structure of watched.

I can upload some code or some pictures if you want to know more.

Lastly, how are people dealing with connections that are lost and providing feedback to the user? I can see if you lose your network connection, it unlinks all the folders/files, but that also happens when you delete something. I was thinking of requesting what is mounted and timing pings to see if it’s connected or not.

It would be good to hear if anyone else is doing the same and what their solution is.

Thanks, Jason

**Comment by paulmillr** [comment]
Well there's `add`, `getWatched` and there's `unwatch`. You can reuse using them.

If you need any particular manipulation methods that are currently not here, let me know which ones.
